woocommerce/google-listings-and-ads
# Integrate Proxied Request Review Status

### Changes proposed in this Pull Request:

Related to GLA Request Review Project 

After finishing the API integration in WCS (already in vipgo) the integration of the middleware as well as the logic fr merging the responses was needed.

This PR adds the next to the feature:

- Proxy request in the Middleware
- Adapt RequestReviewController route method
- Add a new Helper RequestReviewStatuses with the logic to merge the different regionStatues of the 2 programs (free and ads) 
- Updated tests

For now. The logic for merging the response works like this:
We check if any region status is elegible for review, if so, we set `reviewAllowed` as true.
If any region status has Disapproved or Warning state, we add the issues in the `issues` field  
For the account `status` it's still some discussion but for now is build with a priority system, having the worst status as the max priority.

APPROVED
ONBOARDING
PENDING_REVIEW
UNDER_REVIEW
WARING
DISAPPROVED

So for example, if some region is DISAPPROVED the status of the response will be DISAPPROVED doesn't matter if others are in UNDER_REVIEW because DISAPPROVED as more priority. 

### Detailed test instructions:

1. Checkout this branch
2. Go to GLA Product Feed 
3. Check the network panel and you should see a request to `/wc/gla/mc/review`
4. In the response you should see the fields  `issues`, `status` and `reviewAllowed`. The values will depend actually on your GLA account.
